lib/pbench/server/api/resources/users_api.py

```diff
@@ -441,48 +416,47 @@ def put(self, username):
         # TODO: Check if the user has the right privileges
         if not verified:
             self.logger.warning(
-                "Username retrieved from the auth token {} is different from the username provided",
-                self.auth.token_auth.current_user().id,
+                "User {} is not authorized to delete user {}.", user.username, username,
             )
             abort(
-                403, message="Authentication token does not belong to the current user"
+                403,
+                message=f"Not authorized to update information about user {username}",
             )
 
-        # Check if the user payload contain fields that are either non-updatabale or
+        # Check if the user payload contain fields that are either protected or
         # are not present in the user db. If any key in the payload does not match
         # with the column name we will abort the update request.
-        non_existent = list(set(post_data.keys()) - set(User.__table__.columns.keys()))
-        non_updatable = list(
-            set(post_data.keys()).intersection(set(User.get_protected()))
+        non_existent = set(post_data.keys()).difference(
+            set(User.__table__.columns.keys())
         )
-        if non_updatable:
-            for field in non_updatable:
-                if getattr(user, field) != post_data[f"{field}"]:
-                    self.logger.warning(
-                        "Either provided values to update the user does not adhere to the user model "
-                        "datatype or user attempted to update the protected properties."
-                    )
-                    abort(400, message="Invalid data in update request payload")
         if non_existent:
             self.logger.warning(
                 "User trying to update fields that are not present in the user database. Fields: {}",
                 non_existent,
             )
             abort(400, message="Invalid fields in update request payload")
+        protected = set(post_data.keys()).intersection(set(User.get_protected()))
+        for field in protected:
+            if getattr(user, field) != post_data[field]:
+                self.logger.warning(
+                    "User trying to update the non-updatable fields. {}: {}",
+                    field,
+                    post_data[field],
+                )
+                abort(400, message="Invalid data in update request payload")
```

Comment:
We don't want to be helpful (or charitable) and tell the requestor _which_ data is invalid?  (I mean, they _can_ figure it out by trial and error, so why not save them the headache?)

(And, you like `400` here better than `403`?)

Reply:
fixed